### ISSUE TITLE
Links to an assetProxy from within content are not evaluated

### ISSUE THREAD

**Issue by daniellienert** [neos/neos-development-collection#2356]
### Description

On the demo site, link to an asset proxy. 

#### Expected behavior

In the frontend, I get a link to the asset proxy.

#### Actual behavior

I get an assetproxy:// link

### Affected Versions

Neos: dev-master

UI:dev-master

**Comment by dimaip** [comment]
Probably needs to be tackled here, right? https://github.com/neos/neos-development-collection/blob/7bf184dfcda55662f6cb73687eba29848a047705/Neos.Neos/Classes/Fusion/ConvertUrisImplementation.php#L119

**Comment by aertmann** [comment]
Also here:
https://github.com/neos/neos-development-collection/blob/master/Neos.Neos/Classes/Service/LinkingService.php#L68
https://github.com/neos/neos-development-collection/blob/master/Neos.Neos/Classes/Service/LinkingService.php#L206

Potentially also here:
https://github.com/neos/neos-development-collection/blob/master/Neos.Neos/Classes/Domain/Service/NodeShortcutResolver.php#L59

**Comment by aertmann** [comment]
Just wondering why this issue appears now and not before, nothing changed from 4.2.4 to 4.2.5 to cause it as far as I can see. So I'm still suspecting the new UI.

**Comment by aertmann** [comment]
After a bit more investigation I can say that it is indeed the new UI causing issues.

The 1.x branch works as this was reverted https://github.com/neos/neos-ui/commit/6bd1cfffefa6c9ef4ef5b969e7ba6fcd28712948

The `assetProxy://` comes from https://github.com/neos/neos-ui/blob/master/packages/neos-ui-backend-connector/src/Endpoints/index.ts#L331

So it's currently broken due to the new UI supporting the `assetProxy://`, but not all parts of linking does in Neos. Seems like one of those issues caused by different versioning schemes, but even the master of Neos doesn't support it and there's no pending pull request adding support for it.

Any suggestions how to proceed? I'd say we should fix it as a bugfix as it can easily be done in a backwards compatible way.

**Comment by daniellienert** [comment]
We definitely need to fix that in a bug fix version, as all in-content-links generated by the 3.0 of the UI to assets are broken.

**Comment by dimaip** [comment]
Dear @robertlemke, I reverted https://github.com/neos/neos-ui/pull/2161 for now, as it totally breaks even linking to local assets (they all end up with assetProxy).
Let's bring it back in when it's fully supported and tested.

**Comment by kitsunet** [comment]
Good catch, you are probably right that it was never solved for inline links.
Maybe a chance to finally make the backend side configurable (eg. custom schemes to link to things)

**Comment by kitsunet** [comment]
Hi all, see https://github.com/neos/neos-ui/pull/2385 as a fix for that. This adds the asset source support again, but fixes linking so that you cannot link to any non imported asset (which makes sense).

**Comment by kitsunet** [comment]
I changed this to a feature, as it would definitely be nice to have but at this point no such links should be created therefore it's not a bug but simply a missing feature that we cannot link to proxies.

**Comment by kitsunet** [comment]
To clarify, the necessary feature here would be to import the asset proxy just as it happens with images when you select one and then create the link on the imported (proxy) asset.
